src/bio/bio_xstream.c

```diff
@@ -113,12 +112,13 @@ bio_spdk_env_init(void)
 	 */
 
 	if (bio_nvme_configured(SMD_DEV_TYPE_MAX)) {
-		rc = bio_add_allowed_alloc(nvme_glb.bd_nvme_conf, &opts);
+		rc = bio_add_allowed_alloc(nvme_glb.bd_nvme_conf, &opts, &roles);
 		if (rc != 0) {
 			D_ERROR("Failed to add allowed devices to SPDK env, "DF_RC"\n",
 				DP_RC(rc));
 			goto out;
 		}
+		nvme_glb.bd_nvme_roles = roles;
```

Comment:
I'm wondering if we should query the daos_md_backend (see umempobj_setting_init()) and report error earlier if contradicting configuration is detected. For example, meta role is configured in NVMe conf but md backend is set to PMEM, or md backend is set to BMEM/ADMEM but meta role isn't configured in NVMe conf.

That might require a new exported API from umem to query the md backend, and could be done in a follow-on patch.

Reply:
sure